adapta/storage/models/aws.py

```diff
@@ -30,22 +30,34 @@ class S3Path(DataPath):
 
     def to_uri(self) -> str:
         """
-        Not yet implemented in S3Path
+        Converts the S3Path to a URI.
+         :return: URI path
         """
-        raise NotImplementedError
+        if not self.bucket or not self.path:
+            raise ValueError("Bucket and path must be defined")
+
+        return f"s3://{self.bucket}/{self.path}"
 
     def base_uri(self) -> str:
         """
-        Not yet implemented in S3Path
+        Returns the base URI of the S3Path.
+        :return: URI path
         """
-        raise NotImplementedError
+        if not self.bucket:
+            raise ValueError("Bucket must be defined")
+
+        return f"s3://{self.bucket}"
 
     @classmethod
-    def from_uri(cls, url: str) -> "DataPath":
+    def from_uri(cls, url: str) -> "S3Path":
         """
-        Not yet implemented in S3Path
+        Creates an S3Path from a URI.
+        :return: S3Path path
         """
-        raise NotImplementedError
+        assert url.startswith("s3://"), "URI should start with s3://"
```

Comment:
http, not hdfs

Reply:
Fixed in  [11247c6](https://github.com/SneaksAndData/adapta/pull/437/commits/11247c610a0a01c7dc2a7e504107bd99fcfda553)  [11247c6](https://github.com/SneaksAndData/adapta/pull/437/commits/11247c610a0a01c7dc2a7e504107bd99fcfda553)